sycl/include/CL/sycl/interop_handle.hpp

```diff
@@ -0,0 +1,117 @@
+  typename std::enable_if<accessTarget != access::target::host_buffer,
+                          cl_mem>::type
+  get_native_mem(const accessor<dataT, dimensions, accessmode, accessTarget,
+                                isPlaceholder> &Acc) const {
+#ifndef __SYCL_DEVICE_ONLY__
+    // employ reinterpret_cast instead of static_cast due to cycle in includes
+    // involving CL/sycl/accessor.hpp
+    auto *AccBase = const_cast<detail::AccessorBaseHost *>(
+        reinterpret_cast<const detail::AccessorBaseHost *>(&Acc));
+    return getMemImpl(detail::getSyclObjImpl(*AccBase).get());
+#else
+    (void)Acc;
+    // we believe this won't be ever called on device side
+    return static_cast<cl_mem>(0x0);
+#endif
+  }
+
+  template <typename dataT, int dimensions, access::mode accessmode,
+            access::target accessTarget, access::placeholder isPlaceholder>
+  typename std::enable_if<accessTarget == access::target::host_buffer,
+                          cl_mem>::type
+  get_native_mem(const accessor<dataT, dimensions, accessmode, accessTarget,
+                                isPlaceholder> &) const {
+    throw invalid_object_error("Getting memory object out of host accessor is "
+                               "not allowed",
+                               PI_INVALID_MEM_OBJECT);
+  }
+
+  /// Returns an underlying OpenCL queue for the SYCL queue used to submit the
+  /// command group, or the fallback queue if this command-group is re-trying
+  /// execution on an OpenCL queue. The OpenCL command queue returned is
+  /// implementation-defined in cases where the SYCL queue maps to multiple
+  /// underlying OpenCL objects. It is responsibility of the SYCL runtime to
+  /// ensure the OpenCL queue returned is in a state that can be used to
+  /// dispatch work, and that other potential OpenCL command queues associated
+  /// with the same SYCL command queue are not executing commands while the host
+  /// task is executing.
+  cl_command_queue get_native_queue() const noexcept { return MQueue; }
+
+  /// Returns an underlying OpenCL device associated with the SYCL queue used
+  /// to submit the command group, or the fallback queue if this command-group
+  /// is re-trying execution on an OpenCL queue.
+  cl_device_id get_native_device() const noexcept { return MDeviceId; }
+
+  /// Returns an underlying OpenCL context associated with the SYCL queue used
+  /// to submit the command group, or the fallback queue if this command-group
+  /// is re-trying execution on an OpenCL queue.
+  cl_context get_native_context() const noexcept { return MContext; }
+
+private:
+  using ReqToMem = std::pair<detail::Requirement *, pi_mem>;
+
+  template <typename DataT, int Dims, access::mode AccMode,
+            access::target AccTarget, access::placeholder isPlaceholder>
+  friend class accessor;
+  friend class detail::ExecCGCommand;
+  friend class detail::DispatchHostTask;
```

Comment:
What's the point of `interop_handle` having these friends?

Reply:
Well, `ExecCGCommand` and `DispatchHostTask` will require to know about `ReqToMem` type. Though, these `friend` lines are not relevant in this exact patch and may be removed.

Reply:
That makes sense, thank you for clarifying. What does `accessor` gain from this friendship?

Reply:
Accessor won't benefit from it.
This change is redundant.
Sorry to confuse.